site/content@beta/backend-apis/core-info.md

```diff
@@ -0,0 +1,155 @@
+---
+weight: 30
+---
+
+# Key concepts
+
+## Chats and threads
+
+By looking at the chat structure, you notice that each **chat** is is divided into **threads**. Every thread contains **events**, for example sent messages. You can think of a **chat** as **a whole conversation**, while **threads** are separate **conversation topics**.
+
+![Chats and Threads](chat_threads.png "chats and threads")
+
+Consider the example of an online store. A customer starts a chat to ask about the shoes he bought. If he hasn't got any previous chat history, **a new chat is started**. Within this chat, **a new thread is created** (conversation topic). The messages he exchanges with the customer service assistant come to the thread as **events**. After solving the problem, the custmer says goodbye and ends the chat, which automatically **closes the thread**. Let's say the customer is rude and leaves without saying goodbye. In this case, the thread closes after 30 minutes of inactivity (time periods are configurable). The chat **doesn't end**, though.
```

Comment:
Napisałaś `says goodbye and ends the chat` a poniżej jest `The chat **doesn't end**, though.` Może jakoś inaczej napisać w tym pierwszym przypadku?

Reply:
No, racja. Trzeba jakoś rozróżnić, że ktoś kończy czata (w sensie rozmowę), a kończenie (albo raczej nie kończenie się) czata w rozumieniu czata, który składa si z threadów itd.

Reply:
and leaves the chat?

Reply:
Yep, _leaves_ will be better.

Reply:
Oh, I know: _After solving the problem, the customer says goodbye and **closes** the chat__ :)